Add a test verifying that |let<newline>let| isn't subjected to ASI, matches LexicalDeclaration, and is then treated as an early error

jswalden opened the pull request:
See the extensive comment in the testcase for explanation and justification of this testcase.

Caveat: I've verified this test passes in a SpiderMonkey implementing the semantics described in this testcase, but I haven't verified all the test262 testcase structure/formatting is correct and doesn't contain any errors.

rwaldron commented:
This needs to be moved into `test/language/statements/let/syntax`

rwaldron commented on test/language/statements/let/let-let-declaration-split-across-two-lines.js:
```diff
@@ -0,0 +1,36 @@
+  because "let\nlet = ..." matches LexicalDeclaration before static semantics
+  are considered.  *Then* 13.3.1.1's static semantics for the LexicalDeclaration
+  just chosen, per 5.3, are validated to recognize the Script as invalid.  Thus
+  the eval script can't be evaluated, and a SyntaxError is thrown.
```
Needs:

`negative: SyntaxError`

rwaldron commented on test/language/statements/let/let-let-declaration-split-across-two-lines.js:
```diff
@@ -0,0 +1,36 @@
+es6id: 13.3
+flags: noStrict
+description: >
+    let: |let let| split across two lines is not subject to automatic semicolon insertion.
```
If this is the case being tested, doesn't the following suffice: 

```js
let
let = 'parsed';
```

And that should throw?

rwaldron commented on test/language/statements/let/let-let-declaration-split-across-two-lines.js:
```diff
@@ -0,0 +1,36 @@
+{
+  let = "threw " + e.name;
+}
+assert.sameValue(let, "threw SyntaxError");
```
If the answer to my question above is yes, then this can just be: 

```js
let
let = 'parsed';
```

With the additional front matter, also noted above.

goyakin replied:
@jswalden, in addition to what @rwaldron said, would you mind adding the same test for `const` as this applies to it too?

rwaldron commented on test/language/statements/let/let-let-declaration-split-across-two-lines.js:
```diff
@@ -0,0 +1,36 @@
+
+/*---
+author: Jeff Walden
+es6id: 13.3
```
13.3.1.1

jswalden commented:
Okay, in theory I responded to everything.  I moved the tests around, simplified the tests into negative: SyntaxError tests, made the section references more specific, and added a test for `const let = "";` (although really it's a pretty separate consideration from `let let;` as `const` is a keyword always and everywhere, so the failure modes are entirely different).  Anything else?

goyakin commented:
Thanks @jswalden.
